modin/experimental/torch/datasets.py

```diff
@@ -0,0 +1,108 @@
+from __future__ import annotations
+
+from typing import Hashable, NoReturn
+
+import numpy as np
+from pandas import DataFrame, Series
+from torch.utils import data as data_utils
+from torch.utils.data import DataLoader, IterableDataset
+
+from modin.pandas import DataFrame as ModinDataFrame
+from modin.pandas import Series as ModinSeries
+
+
+class ModinIterableDataset(IterableDataset):
+    """
+    PandasIterableDataset is responsible for converting a pandas/modin dataframe into a torch-compatible dataset.
+
+    The optional with_index option determines if index would be part of the output. Default is no.
```

Comment:
No need to include this line if you're mentioning it in the __init__ docstring as well.

```suggestion
```